### PR TITLE
Modify impersonation api key permission check.

### PR DESCRIPTION
**Pull request by vadimberezniker**
The API will be called after impersonation is already in effect in which case we can't rely on checking admin group membership.

Instead check the impersonation bit, which can only be set for authenticated admins.

**Related issues**: N/A

### CODE REVIEW

**Review by bduffany — APPROVED** [review]

**Comment by vadimberezniker** [comment]
Will wait for LGTM from @tylerwilliams before merging.

**Review by tylerwilliams — APPROVED** [review]

**Pull request merged**
